store/striplock.go

```diff
@@ -0,0 +1,55 @@
+package store
+
+import (
+	"sync"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+// TODO: move to utils
+type striplock struct {
+	heights    []*sync.RWMutex
+	datahashes []*sync.RWMutex
+}
+
+type multiLock struct {
+	mu []*sync.RWMutex
```

Comment:
Do we need `RWMutex` ? Looks like `Mutex` will be enough.

Reply:
Also, do we need a slice? looks like we only using 2 values at most

Reply:
Read lock is used in the store itself

Reply:
Yes, the slice is needed and there are actually 1024 entries in it

Reply:
Thx